services/app-graphql/src/queries/fetchSubmittedContract.graphql

```diff
@@ -0,0 +1,299 @@
+query fetchSubmittedContract($input: FetchSubmittedContractInput!) {
```

Comment:
I'm very curious how people with more graphql experience feel about this? I created this query because `fetchContract` returns the `draftRevisions` and `draftRates` field that are null for submitted contracts . This was resulting in a `Cannot return null for a non nullable field` for `contract.draftRevision.id`

Reply:
High level my questions are not about the graphql but the types pattern - to me this change could suggest a different mental model for types coming out of the API.  
- Would you envision then that we narrow the `status` field too - so only `SUBMITTED` or `RESUBMITTED` here?
-  What would be the API response then when you try to fetch a submitted contract that does exist by ID in DB, it is just not submitted (maybe its unlocked for example or it's an initial draft)? 
- If we went forward with this kind of type - should it be defined in `codegen.yml` as well? Not sure what defines what should go in that file.

As I write it out, these kinds of questions feel familiar. They have come up before. It makes me wonder are hitting a limitation in our current loose data model? I think we intentionally did not want to add back locked and unlocked types like we used to have...but is there a case for them coming up here?

Reply:
More specific to the error -  @pearl-truss could you tell me how to replicate the ` Cannot return null for a non nullable field for contract.draftRevision.id` issue - just interested to see what you are talking about, it sounds kind of familiar.

Reply:
@haworku You can replicate it by replacing `useFetchSubmittedContractQuery` with `useFetchContractQuery` and using the V1 submit page to create a submission

And those are good questions! I think a major downside to what I have here is the implications for the ReviewSubmit page since that page is for both draft and submitted contracts. I don't yet have ideas on how to address that. And yeah I think it would make sense to narrow down the status field. If the status of the contract is draft or unlock, it could make sense to return an input error. And I'm not super familiar with codegen.yml but after taking a look it, I think it might make sense to include submittedContracts there but I need to look into it more because I'm not sure where this gets used